pkg/schemacompat/schemacompat_test.go

```diff
@@ -186,6 +186,244 @@ func TestCompatibility(t *testing.T) {
 				"existing": {Type: "string"},
 			},
 		},
+	}, {
+		desc: "new has more properties, existing contains number",
+		existing: &apiextensionsv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "number"},
+			},
+		},
+		new: &apiextensionsv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "number"},
+				"new":      {Type: "integer"},
+			},
+		},
+		// LCD is the same as existing.
+		wantLCD: &apiextensionsv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "number"},
+			},
+		},
+	}, {
+		desc: "new has more properties, existing contains integer",
+		existing: &apiextensionsv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "integer"},
+			},
+		},
+		new: &apiextensionsv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "integer"},
+				"new":      {Type: "number"},
+			},
+		},
+		// LCD is the same as existing.
+		wantLCD: &apiextensionsv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "integer"},
+			},
+		},
+	}, {
+		desc: "new has more properties, existing contains XIntOrString",
+		existing: &apiextensionsv1.JSONSchemaProps{
+			Type:         "",
+			XIntOrString: true,
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "integer"},
+			},
+		},
+		new: &apiextensionsv1.JSONSchemaProps{
+			Type:         "",
+			XIntOrString: true,
+			Properties: map[string]apiextensionsv1.JSONSchemaProps{
+				"existing": {Type: "integer"},
```

Comment:
I don't see the new properties

Reply:
Thanks @davidfestal, I added new properties.